### ISSUE TITLE
Battery Life

### ISSUE THREAD

**Issue by qeeqez** [daliansky/XiaoMi-Pro#97]
Does anybody here have a long battery life?

Same brightness and usage:

- Mac: 4.5-5h max
- Windows: 6h normal, 7-8 with Throttlestop
- Linux: 7 -> up to 10h with powertop and other powersaving enabled features

I'm using VoltageShift but I think it does nothing, any suggestions?

**Comment by stevezhengshiqi** [comment]
@qeeqez Hi, have you tried the CPUFriendProvider for battery life in README link?

**Comment by Mateo1234454545** [comment]
Can I ask something ?
CPUFriendDataProvider with Cpufriend will reduce cpu freq when idle. But voltage is not affected. 
Isn't voltage the one which we should care about ?
I mean , even if cpu runs at 0,8 , battery time will be the same if voltage is not reduced.
Am I wrong?

**Comment by qeeqez** [comment]
> @qeeqez Hi, have you tried the CPUFriendDataProvider for battery life in README link?

Yep, 6h (I think up to 7) but Performance is low, so I'm using Balanced CPUFriendDataProvider.

Maybe it's macOS-specific and power states shouldn't change dynamically here, otherwise, Windows could achieve good battery and high scores in Geekbench at once

We need to know how to stay long on battery and have decent performance on AC.

**Comment by stevezhengshiqi** [comment]
@Mateo1234454545 I think voltage is been locked by BIOS and can't be modified by system. In Windows, we have to use XTU or some third-party app to unlock it.

**Comment by Mateo1234454545** [comment]
Yes, I know.
So , my guess is, with voltage locked cpu speed will not affect battery.

**Comment by FallenChromium** [comment]
I believe voltage can be overwritten in BIOS (actially i did so), and results are not impressive, but good

**Comment by fladnaG86** [comment]
hi i managed to undervolt cpu through "voltageshift" getting much lower temperatures and consumption.
I have set offset a -150 for cpu -90 for gpu and -150 for cache memory.
the battery lasts a lot longer, the performance are much higher. 
[voltageshift.zip](https://github.com/daliansky/XiaoMi-Pro/files/2827343/voltageshift.zip)
I have created two scripts, they must be modified with your login password. The voltageshift folder should be copied into downloads.
My xiaomi is the core i5 mx150 with bios 603 and patch for undervolt.
